DXMainClient/Domain/Multiplayer/MapLoader.cs

```diff
@@ -205,8 +211,16 @@ private void CacheCustomMaps(ConcurrentDictionary<string, Map> customMapCache)
             try
             {
                 var jsonData = File.ReadAllText(CUSTOM_MAPS_CACHE);
-            
-                return JsonConvert.DeserializeObject<ConcurrentDictionary<string, Map>>(jsonData) ?? new ConcurrentDictionary<string, Map>();
+
+                var customMapCache = JsonConvert.DeserializeObject<CustomMapCache>(jsonData);
+
+                var customMaps = customMapCache?.Version == CurrentCustomMapCacheVersion && customMapCache.Maps != null
+                    ? customMapCache.Maps : new ConcurrentDictionary<string, Map>();
+
+                foreach (var customMap in customMaps.Values)
+                    customMap.CalculateSHA();
```

Comment:
Calculating the SHA1 hash here for all cached maps and in `LoadCustomMaps` for all local maps effectively makes it so that client calculates the hash twice for all custom maps if they also exist in the cache. I am however unable to discern any performance impact from this, likely due to either operating system or hardware-level optimizations concerning file reading. Because of this and the fact that doing this in any other way would compromise the robustness of using SHA1 hash to identify maps in the first place, I don't believe this requires any action but I thought it was worth noting regardless.